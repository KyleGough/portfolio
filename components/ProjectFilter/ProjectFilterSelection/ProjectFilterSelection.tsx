import React, { KeyboardEvent } from 'react';

interface ProjectFilterSelectionProps {
  onSelect: (a: string) => void;
  value: string;
}

export const ProjectFilterSelection: React.FC<ProjectFilterSelectionProps> = ({
  value,
  onSelect,
}) => {
  const onKeyDown = (value: string, event: KeyboardEvent<HTMLLIElement>) => {
    if (event.key === 'Enter') onSelect(value);
  };

  return (
    <li
      key={value}
      tabIndex={0}
      onClick={() => onSelect(value)}
      onKeyDown={(event: KeyboardEvent<HTMLLIElement>) =>
        onKeyDown(value, event)
      }
      className="hover:text-chip hover:bg-chip-light py-2 px-8 cursor-pointer select-none"
    >
      {value}
    </li>
  );
};
