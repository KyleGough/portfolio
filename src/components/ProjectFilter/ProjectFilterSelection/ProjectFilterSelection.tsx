import React from 'react';

interface ProjectFilterSelectionProps {
  value: string;
  onClick: (a: string) => void;
}

export const ProjectFilterSelection: React.FC<ProjectFilterSelectionProps> = ({
  value,
  onClick,
}) => (
  <li
    key={value}
    onClick={() => onClick(value)}
    className="hover:text-chip hover:bg-chip-light py-2 px-8 cursor-pointer select-none"
  >
    {value}
  </li>
);
