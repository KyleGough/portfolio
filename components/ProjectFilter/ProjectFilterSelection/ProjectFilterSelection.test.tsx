import { render, screen } from '@testing-library/react';
import React from 'react';

import { ProjectFilterSelection } from './ProjectFilterSelection';

describe('ProjectFilterSelection component', () => {
  it('renders', () => {
    const { container } = render(
      <ProjectFilterSelection value="Web" onSelect={() => jest.fn()} />
    );

    expect(screen.getByText('Web')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
