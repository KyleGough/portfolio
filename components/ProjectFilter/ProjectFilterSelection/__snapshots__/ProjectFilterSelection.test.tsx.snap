// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFilterSelection component renders 1`] = `
<div>
  <li
    class="hover:text-chip hover:bg-chip-light py-2 px-8 cursor-pointer select-none"
    tabindex="0"
  >
    Web
  </li>
</div>
`;
